### PR TITLE
Updating the documentation (fixing typos)

### PR DESCRIPTION
**Pull request by garanews**
### Type of Change

- Updating the documentation (fixing typos)

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [ ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ ] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/emojicode/concepts/repeat-while/repeat-while.md
Comment (SSwiniarski) [inline content/emojicode/concepts/repeat-while/repeat-while.md:3]:
```suggestion
Description: 'The 🔁 statement repeats a code block while as long as the given condition is 👍 (true). And if the condition is never 👍 (false), the code block will be ignored and skipped. Due to the ease of use of the 🔂 (for-in) statement, 🔁 is seldom used. emojic 🏁 🍇 🔁 👍 🍇 😀 🔤It goes on and on and on🔤❗️ 🍉'
```

File: content/emojicode/concepts/repeat-while/repeat-while.md
Comment (SSwiniarski) [inline content/emojicode/concepts/repeat-while/repeat-while.md:18]:
```suggestion
Due to the ease of use of the 🔂 (for-in) statement, 🔁 is seldom used.
```

File: content/general/concepts/cache/cache.md
Comment (SSwiniarski) [inline content/general/concepts/cache/cache.md:3]:
```suggestion
Description: 'A cache is data stored locally in an application for faster retrieval.'
```

File: content/general/concepts/iot/iot.md
Comment (SSwiniarski) [inline content/general/concepts/iot/iot.md:3]:
```suggestion
Description: 'Internet of Things (IoT) is a term used to describe hardware which can be connected to the internet in order to provide a service.'
```

File: content/general/concepts/machine-code/machine-code.md
Comment (SSwiniarski) [inline content/general/concepts/machine-code/machine-code.md:3]:
```suggestion
Description: 'Machine code (also known as machine language or native code) is a low level programming language in the form of hexadecimal or binary instructions that execute computer programs on the computers CPU.'
```

File: content/general/concepts/processor/processor.md
Comment (SSwiniarski) [inline content/general/concepts/processor/processor.md:3]:
```suggestion
Description: 'A processor is the electric circuitry found in computer hardware which is responsible for executing instruction sets derived from programs that have been converted into machine code.'
```

File: content/general/concepts/ram/ram.md
Comment (SSwiniarski) [inline content/general/concepts/ram/ram.md:3]:
```suggestion
Description: 'RAM is an acronym for Random Access Memory, which is a mechanism for storing and retrieving information in random memory units as opposed to other formats that store data in specific memory locations.'
```

File: content/general/concepts/ssl/ssl.md
Comment (SSwiniarski) [inline content/general/concepts/ssl/ssl.md:3]:
```suggestion
Description: 'SSL stands for Secure Socket Layer, and is a cryptographic protocol for securing computer networks.'
```

File: content/javascript/concepts/arrow-functions/arrow-functions.md
Comment (SSwiniarski) [inline content/javascript/concepts/arrow-functions/arrow-functions.md:3]:
```suggestion
Description: 'Arrow function expressions were introduced in ES6. These expressions are a clean and concise alternative to the traditional function syntax.'
```

File: content/javascript/concepts/npm/npm.md
Comment (SSwiniarski) [inline content/javascript/concepts/npm/npm.md:3]:
```suggestion
Description: 'Node Package Manager (NPM) is a large collection of JavaScript packages. This makes adding complexity and functionality to JavaScript project seamless.'
```

File: content/php/concepts/variables/variables.md
Comment (SSwiniarski) [inline content/php/concepts/variables/variables.md:3]:
```suggestion
Description: 'Variables store data for later use, and allow their contents to be updated or changed.'
```

File: content/react/concepts/virtual-dom/virtual-dom.md
Comment (SSwiniarski) [inline content/react/concepts/virtual-dom/virtual-dom.md:3]:
```suggestion
Description: 'In React, Virtual DOM is a conceptual representation of the actual DOM object, like a lightweight copy. A virtual DOM object has the same properties as a real DOM object, but it lacks the real thing’s power to directly change what’s on the screen.'
```

File: content/ruby/concepts/casting/casting.md
Comment (SSwiniarski) [inline content/ruby/concepts/casting/casting.md:3]:
```suggestion
Description: 'In Ruby, the language comes with a variety of building typecasting methods for converting values from one data type to another.'
```

File: content/sql/concepts/window-functions/window-functions.md
Comment (SSwiniarski) [inline content/sql/concepts/window-functions/window-functions.md:3]:
```suggestion
Description: 'A window function performs a calculation over the inputted column and then returns the calculated value for each row.'
```

File: content/swift/concepts/arrays/arrays.md
Comment (SSwiniarski) [inline content/swift/concepts/arrays/arrays.md:3]:
```suggestion
Description: 'Arrays are ordered collections of values. In other words, an array is an ordered list of values, all of the same type, contained in the same structure.'
```

File: content/swift/concepts/comments/comments.md
Comment (SSwiniarski) [inline content/swift/concepts/comments/comments.md:3]:
```suggestion
Description: 'Comments in represent text that is useful to the programmer or development team.'
```

File: content/swift/concepts/variables/variables.md
Comment (SSwiniarski) [inline content/swift/concepts/variables/variables.md:3]:
```suggestion
Description: 'The Swift programming language allows you to store values in two types of containers, var for variables and let for constants.'
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@garanews, This was large. I'll defer to other maintainers if they want to have you split this into smaller PRs. I did a review as is. Most of my comments were to shorten the descriptions in the existing articles as they are longer than the 1-2 sentence length we prefer now. 

@Dusch4593, @yangc95, @KTom101 -- does this conflict with any outstanding PRs?

File: content/swift/concepts/arrays/arrays.md
Comment (yangc95) [reply content/swift/concepts/arrays/arrays.md:3]: @garanews @SSwiniarski I am resolving these swift entries since I am currently working on edits to all of swift

File: content/sql/concepts/window-functions/window-functions.md
Comment (yangc95) [inline content/sql/concepts/window-functions/window-functions.md:19]: @garanews please revert all the changes for Swift entries. There's currently a larger PR our for Swift entry edits here: https://github.com/Codecademy/docs/pull/929

Comment (yangc95) [reply content/sql/concepts/window-functions/window-functions.md:19]: You're free to comment on the above PR if you see anything that may need tweaking

**Review by yangc95 — CHANGES_REQUESTED** [review]
@garanews I left some comments but I will also be taking a more thorough look into this PR as well since this PR affects more than the recommended file number changes.

**Comment by Dusch4593** [comment]
@yangc95 @SSwiniarski I'm actually going to recommend that @garanews resubmit these typo fixes in the form of multiple pull requests.

## My Concerns

The rationale is that this PR presents two issues of note:

- Because this PR covers 100+ files spanning multiple concepts and terms (some of which are already being edited in other open PRs). We technically ask our contributors to [limit to 3 files per PR](https://github.com/Codecademy/docs/pull/1032) but I understand why these edits cover more than 3. My main concern with this point is that if multiple open PRs are potentially making changes to the same file(s) on the same line(s), then [merge conflicts](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/about-merge-conflicts) could appear in multiple PRs.
- These edits were pushed from [your fork's `main` branch](https://github.com/garanews/docs-1), which we discourage contributors from doing. The reason is that the `main` branch of your forked repo should always be in sync with the upstream [the `main` branch under `Codecademy/docs`](https://github.com/Codecademy/docs). Edits and new content (including new files and folders) should be done in a separate PR branch.

## What To Do Going Forward

I'm going to close this PR and ask that you submit these typo fixes in the form of _multiple pull requests (one for each Docs topic)_.

For example, you could make a new Git branch called `typos-c-topic` that addresses all typos in entries found under [C](https://github.com/Codecademy/docs/tree/main/content/c), another one called `typos-javascript-topic` for fixes to entries under [JavaScript](https://github.com/Codecademy/docs/tree/main/content/javascript).

Please do not make changes to the following content as they are being handled in other open PRs:

- All Swift-related entries.
- Under C++
  - content/cpp/concepts/vectors/vectors.md
- Under General:
  - content/general/concepts/angular/angular.md
  - content/general/concepts/back-end/back-end.md
  - content/general/concepts/framework/framework.md
  - content/general/concepts/next-js/next-js.md
-Under Git:
  - content/git/git.md
  - content/git/concepts/commit/commit.md

Let us know if you have any questions about any of this. Happy to help 😄

**Pull request closed without merging**
